packages/core/src/validate/run.ts

```diff
@@ -241,13 +243,35 @@ function* getConstructorErrors(contractDef: ContractDefinition, decodeSrc: SrcDe
   }
 }
 
+function initCaches() {
```

Comment:
I think I would just change this to a function that just creates and returns a single cache. Then we just call it once for each cache we want.